internal/k8s/controller.go

```diff
@@ -277,20 +277,27 @@ func NewLoadBalancerController(input NewLoadBalancerControllerInput) *LoadBalanc
 		lbc.externalDNSController = ed_controller.NewController(ed_controller.BuildOpts(context.TODO(), lbc.namespaceList, lbc.recorder, lbc.confClient, input.ResyncPeriod, isDynamicNs))
 	}
 
+	glog.V(3).Infof("Nginx Ingress Controller has class: %v", input.IngressClass)
+
+	collectorConfig := telemetry.CollectorConfig{
+		K8sClientReader:       input.KubeClient,
+		CustomK8sClientReader: input.ConfClient,
+		Namespaces:            lbc.namespaceList,
+	}
+
 	// NIC Telemetry Reporting
 	if input.EnableTelemetryReporting {
 		lbc.telemetryChan = make(chan struct{})
 		collector, err := telemetry.NewCollector(
-			telemetry.WithTimePeriod(input.TelemetryReportingPeriod),
+			collectorConfig,
+			telemetry.WithTimePeriod("24h"),
```

Comment:
since the new collector takes a config struct, it may be worth to consider adding the time period param as a field. this way `NewCollector` would take only one param - the config struct. less paperwork.